cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,283 @@ func TestConstructor(t *testing.T) {
-		mc.Close()
+func TestGroupConsumer(t *testing.T) {
+	config := sarama.NewConfig()
+	config.ClientID = t.Name()
+	config.Version = sarama.V2_0_0_0
+	config.Consumer.Return.Errors = true
+	config.Consumer.Group.Rebalance.Retry.Max = 2
+	config.Consumer.Offsets.AutoCommit.Enable = false
+
+	msg1, err := newSampleSpan(model.NewTraceID(1, 2), model.NewSpanID(1))
+	require.NoError(t, err)
+	msg2, err := newSampleSpan(model.NewTraceID(2, 3), model.NewSpanID(2))
+	require.NoError(t, err)
+
+	broker0 := sarama.NewMockBroker(t, 0)
+	broker0.SetHandlerByMap(map[string]sarama.MockResponse{
+		"MetadataRequest": sarama.NewMockMetadataResponse(t).
+			SetBroker(broker0.Addr(), broker0.BrokerID()).
+			SetLeader(topic, 0, broker0.BrokerID()),
+		"OffsetRequest": sarama.NewMockOffsetResponse(t).
+			SetOffset(topic, 0, sarama.OffsetOldest, 0).
+			SetOffset(topic, 0, sarama.OffsetNewest, 1),
+		"FindCoordinatorRequest": sarama.NewMockFindCoordinatorResponse(t).
+			SetCoordinator(sarama.CoordinatorGroup, group, broker0),
+		"HeartbeatRequest": sarama.NewMockHeartbeatResponse(t),
+		"JoinGroupRequest": sarama.NewMockSequence(
+			sarama.NewMockJoinGroupResponse(t).SetError(sarama.ErrOffsetsLoadInProgress),
+			sarama.NewMockJoinGroupResponse(t).SetGroupProtocol(sarama.RangeBalanceStrategyName),
+		),
+		"SyncGroupRequest": sarama.NewMockSequence(
+			sarama.NewMockSyncGroupResponse(t).SetError(sarama.ErrOffsetsLoadInProgress),
+			sarama.NewMockSyncGroupResponse(t).SetMemberAssignment(
+				&sarama.ConsumerGroupMemberAssignment{
+					Version: 0,
+					Topics: map[string][]int32{
+						topic: {0},
+					},
+				}),
+		),
+		"OffsetFetchRequest": sarama.NewMockOffsetFetchResponse(t).
+			SetOffset(group, topic, 0, 0, "", sarama.ErrNoError).
+			SetOffset(group, topic, 0, 1, "", sarama.ErrNoError).
+			SetError(sarama.ErrNoError),
+		"FetchRequest": sarama.NewMockSequence(
+			sarama.NewMockFetchResponse(t, 1).
+				SetMessage(topic, 0, 0, sarama.StringEncoder(msg1)).
+				SetMessage(topic, 0, 1, sarama.StringEncoder(msg2)),
+			sarama.NewMockFetchResponse(t, 1),
+		),
 	})
-	saramaClusterConsumer.On("MarkPartitionOffset", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
-	return saramaClusterConsumer
-}
 
-func newConsumer(
-	t *testing.T,
-	metricsFactory metrics.Factory,
-	topic string,
-	processor processor.SpanProcessor,
-	consumer consumer.Consumer,
-) *Consumer {
-	logger, _ := zap.NewDevelopment()
-	consumerParams := Params{
-		MetricsFactory:   metricsFactory,
-		Logger:           logger,
-		InternalConsumer: consumer,
-		ProcessorFactory: ProcessorFactory{
-			topic:          topic,
-			consumer:       consumer,
-			metricsFactory: metricsFactory,
-			logger:         logger,
-			baseProcessor:  processor,
-			parallelism:    1,
-		},
+	saramaConsumer, err := sarama.NewConsumerGroup([]string{broker0.Addr()}, group, config)
+	require.NoError(t, err)
+
+	defer func() { _ = saramaConsumer.Close() }()
```

Comment:
What do you think about putting a `require.NoError` check on the returned error here? It seems we expect this to never return an error. Same below.

Reply:
It would be better to keep error detection, currently 'broker' is using 'sarama. NewMockBroker' implementation, will continue to upgrade sarama to the latest version in the future, configuration parameter changes may cause errors


`require.NoError `-> `assert.NoError`

in sarama
---

<img width="806" alt="image" src="https://user-images.githubusercontent.com/8416556/235399592-eb789d9e-b96d-4a98-a4c7-3611e6694ba1.png">


<img width="788" alt="image" src="https://user-images.githubusercontent.com/8416556/235399365-e83daecd-d8f4-41cd-987e-3f957ef2b433.png">